pkg/upgrade/job/job.go

```diff
@@ -310,18 +311,42 @@ func New(plan *upgradeapiv1.Plan, node *corev1.Node, controllerName string) *bat
 		})
 	}
 
+	// Determine if the target node is Windows
+	isWindows := node.Labels["kubernetes.io/os"] == "windows"
+
+	// Initialize with the default kubectl image; can be changed to Windows or other image if needed
+	selectedKubectlImage := KubectlImage
+	if isWindows {
+		// Retrieve the Windows kubectl image from the SYSTEM_UPGRADE_JOB_KUBECTL_IMAGE_WINDOWS environment variable.
+		// This environment variable must be set by customers/operators when performing upgrades on Windows nodes.
+		selectedKubectlImage = os.Getenv("SYSTEM_UPGRADE_JOB_KUBECTL_IMAGE_WINDOWS")
+		if selectedKubectlImage == "" {
+			logrus.Fatal("SYSTEM_UPGRADE_JOB_KUBECTL_IMAGE_WINDOWS is a required environment variable when targeting Windows")
+		}
```

Comment:
Killing the whole controller over this seems like an overreaction. Other plans that don't run against Windows nodes will still work fine, as will any nodes in THIS plan that are not Windows.

Do this env var lookup in the same place we do all the OTHER env var lookups for global settings, and then just fail plan validation if any of the nodes that match the nodeSelector are windows but this variable is not set.